kinto_attachment/utils.py

```diff
@@ -217,3 +217,17 @@ def save_file(content, request, randomize=True, gzipped=False):
     })
 
     return attachment
+
+
+def setting_value(request, name, default):
+    value = request.registry.settings.get('attachment.{}'.format(name), default)
+    try:
+        bid = '/buckets/{bucket_id}'.format_map(request.matchdict)
+        if bid in request.registry.attachment_resources:
+            value = request.registry.attachment_resources[bid][name]
+        cid = '/buckets/{bucket_id}/collections/{collection_id}'.format_map(request.matchdict)
+        if cid in request.registry.attachment_resources:
+            value = request.registry.attachment_resources[cid][name]
+    except KeyError:
+        pass
+    return value
```

Comment:
I didn't get what you mean...

The `KeyError` is for the format part, when the matchdict does not have `bucket_id` or `collection_id`

Reply:
Oh... that didn't even occur to me. I thought you were catching `KeyError` instead of checking for existence in `request.registry.attachment_resources[cid][name]`. I don't like catching `KeyError` over so many lines of code, it obscures intent, and I definitely don't like catching `KeyError` over `format_map`, that seems very surprising.

Reply:
Totally agree, thanks for your feedback :)